### PR TITLE
add Wlogical-op, cleanup platform compiler flags, and fix missing initializers

### PR DESCRIPTION
**Pull request by dagar**

### CODE REVIEW

File: src/drivers/bmi160/bmi160.cpp [hunk 1]
```diff
@@ -1290,7 +1290,7 @@ BMI160::measure()
 	grb.temperature = _last_temperature;
 
 	/* return device ID */
-	grb.device_id = _gyro->_device_id.devid;
+	grb.device_id = _gyro->_device_id.devid;;
```
Comment (LorenzMeier) [inline src/drivers/bmi160/bmi160.cpp]: This looks wrong.

**Comment by LorenzMeier** [comment]
@bkueng Can you test on RPI? @ChristophTobler Can you test on Snapdragon? @PX4TestFlights Can you please test on all airframes?

**Comment by dagar** [comment]
Added sign-compare checks (https://github.com/PX4/Firmware/pull/7148) to simplify testing.

**Comment by santiago3dr** [comment]
arms and throttle okay but pitch and roll are non responsive

pixracer
http://logs.px4.io/plot_app?log=46615beb-d905-4853-8e87-96323c5e3425

pixhawk 1
http://logs.px4.io/plot_app?log=41867de7-d3fe-40a0-9376-c81385a36b62

**Comment by ChristophTobler** [comment]
build and upload worked, but I didn't test it because of https://github.com/PX4/Firmware/issues/7362

File: src/systemcmds/tests/test_param.c [hunk 2]
```diff
@@ -49,7 +49,6 @@
 /**
```
Comment (bkueng) [inline src/systemcmds/tests/test_param.c]: This comment can be removed as well

**Comment by bkueng** [comment]
Bench tested on RPi, works

**Comment by LorenzMeier** [comment]
@dagar Looks like we missed to merge in time, sorry! Would you mind rebasing?

**Comment by dagar** [comment]
No problem, sprawling PRs are probably a bad idea.

I'll split this up into a few different PRs.

**Comment by dagar** [comment]
Largely replaced by https://github.com/PX4/Firmware/pull/7973, https://github.com/PX4/Firmware/pull/7974, https://github.com/PX4/Firmware/pull/7975, https://github.com/PX4/Firmware/pull/7976

**Pull request closed without merging**